Source/WebCore/platform/network/ResourceResponseBase.h

```diff
@@ -71,11 +71,6 @@ class ResourceResponseBase {
     static bool isRedirectionStatusCode(int code) { return code == 301 || code == 302 || code == 303 || code == 307 || code == 308; }
 
     struct CrossThreadData {
-        CrossThreadData(const CrossThreadData&) = delete;
-        CrossThreadData& operator=(const CrossThreadData&) = delete;
-        CrossThreadData() = default;
-        CrossThreadData(CrossThreadData&&) = default;
-
```

Comment:
If a copy is kept and the cross thread data is sent to another thread, we break the assumption.
Removing the copy constructor protects from such mistakes.
We can probably store ResourceResponse in `CacheStorageMemoryStore `, I don't think the potential perf optimization is worth it.

Reply:
`CacheStorageMemoryStore` is going to be accessed from different threads (since it's accessed on WorkQueue). 
A possible solution is to store `CrossThreadData` in `CacheStorageMemoryStore`, but every time we need to read it, we take it out, convert it to `ResourceResponse`, and call `crossThreadData()` twice on the `ResourceResponse`  to create two `CrossThreadData` -- one for result and one for storage.
And we do the conversion every time we need a copy of CrossThreadData. Is this better than current approach? If so I will make the change.

Reply:
If we store `ResourceResponse` and the only thing we are doing is creating a CrossThreadData from it, this should be safe. Maybe we can store `const ResourceResponse` to help ensuring it is never mutated. 
Or you can add a specific method to clone a CrossThreadData.

Reply:
hmm it should be safe but I think `AtomString` can only be read from the thread that creates it?

If we still need to store `CrossThreadData` in `CacheStorageRecord` and mark `CrossThreadData` as non-copyable, then `CacheStorageRecord` will need to be non-copyable and the code might be more complicated, e.g. we will need to add more code like constructor (from moving `CrossThreadData`) and `copy()`. Do you want me to make the change?

Reply:
Updated the patch by keeping `CrossThreadData` non-copyable.